ERCS/erc-7818.md

```diff
@@ -0,0 +1,162 @@
+     */
+    function balanceOf(
+        address account,
+        uint256 identifier
+    ) external view returns (uint256);
+
+    /**
+     * @dev Retrieves the current epoch of the contract.
+     * @return uint256 The current epoch of the token contract,
+     * often used for determining active/expired states.
+     */
+    function epoch() external view returns (uint256);
+
+    /**
+     * @dev Retrieves the duration a token remains valid.
+     * @return uint256 The validity duration.
+     * @notice `duration` "MUST" specify the token's validity period.
+     * The implementing contract "SHOULD" clearly document,
+     * whether the unit is blocks or time in seconds.
+     */
+    function duration() external view returns (uint256);
+
+    /**
+     * @dev Checks whether a specific `identifier` is expired.
+     * @param identifier The Identifier "MAY" represents an epoch, round, period, or token identifier.
+     * @return bool True if the token is expired, false otherwise.
+     * @notice Implementing contracts "MUST" define the logic for determining expiration,
+     * typically by comparing the current `epoch()` with the given `identifier`.
+     */
+    function expired(uint256 identifier) external view returns (bool);
+
+    /**
+     * @dev Transfers a specific `identifier` and value to a recipient.
+     * @param to The recipient address.
+     * @param identifier The Identifier "MAY" represents an epoch, round, period, or token identifier.
+     * @param value The amount to transfer.
+     * @return bool True if the transfer succeeded, false or reverted if give `identifier` it's expired.
+     * @notice The transfer "MUST" revert if the token `identifier` is expired.
+     */
+    function transfer(
+        address to,
+        uint256 identifier,
+        uint256 value
+    ) external returns (bool);
+
+    /**
+     * @dev Transfers a specific `identifier` and value from one account to another.
+     * @param from The sender's address.
+     * @param to The recipient's address.
+     * @param identifier The Identifier "MAY" represents an epoch, round, period, or token identifier.
+     * @param value The amount to transfer.
+     * @return bool True if the transfer succeeded, false or reverted if give `identifier` it's expired.
+     * @notice The transfer "MUST" revert if the token `identifier` is expired.
+     */
+    function transferFrom(
+        address from,
+        address to,
+        uint256 identifier,
+        uint256 value
+    ) external returns (bool);
+}
+```
+
+### Behavior specification
+
+- `balanceOf` **MUST** return the total balance of tokens held by an account that are still valid (i.e., have not expired). This includes any tokens associated with specific periods, epochs, or other identifiers, provided they remain within their validity duration. Expired tokens **MUST NOT** be included in the returned balance, ensuring that only actively usable tokens are reflected in the result.
```

Comment:
This is ambiguous. Do you mean _just_ the newly introduced `balanceOf` or both this one and the original?